### PR TITLE
Change DockerTestAuth from functional to integ

### PR DESCRIPTION
**Pull request by shubham2892**
### Summary

There is already an integration test `TestDockerCfgAuth` which tests the same functionality as `TestDockerAuth`. `TestDockerAuth` additionally scrapes through the log to check it does not contain any authentication detail.

This PR adds a function which when added to an integration tests stores the log in the same way as stored by functional tests and adds the log check to `TestDockerCfgAuth`.

### Testing

New tests cover the changes: yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/common_integ_test.go [hunk 1]
```diff
@@ -56,6 +60,43 @@ func createTestTask(arn string) *apitask.Task {
...
+
+func loggerConfigIntegrationTest(logfile string) string {
+	config := `
+	<seelog type="asyncloop" minlevel="debug">
+		<outputs formatid="main">
+			<console />`
+	if logfile != "" {
+		config += `<rollingfile filename="` + logfile + `/ecs-agent-log.log" type="date"
+			 datepattern="2006-01-02-15" archivetype="none" maxrolls="24" />`
+	}
```
Comment (yunhee-l) [inline agent/engine/common_integ_test.go]: nit: would using sprintf make this a bit neater?

Comment (shubham2892) [reply agent/engine/common_integ_test.go]: Yes will use that.

File: agent/engine/common_integ_test.go [hunk 2]
```diff
@@ -56,6 +61,40 @@ func createTestTask(arn string) *apitask.Task {
 	}
 }
 
+func setupIntegTestLogs(t *testing.T) string {
+	// Create a directory for storing test logs.
+	testLogDir, err := ioutil.TempDir("", "ecs-integ-test")
+	require.NoError(t, err, "Unable to create directory for storing test logs")
+
+	logger, err := log.LoggerFromConfigAsString(loggerConfigIntegrationTest(testLogDir))
+	if err != nil {
```
Comment (sharanyad) [inline agent/engine/common_integ_test.go]: can use `assert` here and below

File: agent/functional_tests/tests/functionaltests_unix_test.go [hunk 3]
```diff
@@ -156,56 +156,6 @@ func TestCommandOverrides(t *testing.T) {
 	assert.Equal(t, 21, exitCode, fmt.Sprintf("Expected exit code of 21; got %d", exitCode))
 }
 
-func TestDockerAuth(t *testing.T) {
-	agent := RunAgent(t, &AgentOptions{
-		ExtraEnvironment: map[string]string{
-			"ECS_ENGINE_AUTH_TYPE": "dockercfg",
```
Comment (sharanyad) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: with this migration, we essentially lose testing of these env vars?

Comment (shubham2892) [reply agent/functional_tests/tests/functionaltests_unix_test.go]: We are still setting the same flags with these values but in task definition instead os environment variables.

**Pull request merged**